File: packages/core-go/routing/result.go
package routing

import (
	"encoding/json"
	"fmt"
	"strconv"

	"github.com/stellar-address-kit/core-go/address"
)

type MemoType string

const (
	MemoTypeNone   MemoType = "none"
	MemoTypeID     MemoType = "id"
	MemoTypeText   MemoType = "text"
	MemoTypeHash   MemoType = "hash"
	MemoTypeReturn MemoType = "return"
)

type RoutingID struct {
	raw string
}

func (r *RoutingID) UnmarshalJSON(data []byte) error {
	if r == nil {
		return fmt.Errorf("routing id: UnmarshalJSON on nil receiver")
	}

	if string(data) == "null" {
		r.raw = ""
		return nil
	}

	var id string
	if len(data) > 0 && data[0] == '"' {
		if err := json.Unmarshal(data, &id); err != nil {
			return fmt.Errorf("routing id: invalid quoted value: %w", err)
		}
	} else {
		id = string(data)
	}

	parsed, err := strconv.ParseUint(id, 10, 64)
	if err != nil {
		return fmt.Errorf("routing id: invalid uint64 value %q: %w", id, err)
	}

	r.raw = strconv.FormatUint(parsed, 10)
	return nil
}

func (r *RoutingID) String() string {
	if r == nil {
		return ""
	}
	return r.raw
}

func (r *RoutingID) Uint64() (uint64, error) {
	if r == nil {
		return 0, strconv.ErrSyntax
	}
	return strconv.ParseUint(r.raw, 10, 64)
}

func NewRoutingID(s string) *RoutingID {
	return &RoutingID{raw: s}
}

type RoutingResult struct {
	DestinationBaseAccount string
	RoutingID              *RoutingID
	RoutingSource          string // "muxed" | "memo" | "none"
	Warnings               []address.Warning
	DestinationError       *DestinationError
}

type DestinationError struct {
	Code    address.ErrorCode
	Message string
}

File: packages/core-dart/lib/src/routing/result.dart
import '../address/codes.dart';

class RoutingInput {
  final String destination;
  final String memoType;
  final String? memoValue;
  final String? sourceAccount;

  RoutingInput({
    required this.destination,
    required this.memoType,
    this.memoValue,
    this.sourceAccount,
  });
}

class RoutingResult {
  final String? destinationBaseAccount;
  final String? routingId; // decimal uint64 string — spec level
  final String routingSource; // 'muxed' | 'memo' | 'none'
  final List<Warning> warnings;
  final DestinationError? destinationError;

  RoutingResult({
    this.destinationBaseAccount,
    this.routingId,
    required this.routingSource,
    required this.warnings,
    this.destinationError,
  });

  BigInt? get routingIdAsBigInt =>
      routingId != null ? BigInt.parse(routingId!) : null;
}

class DestinationError {
  final String code; // ErrorCode constant
  final String message;

  DestinationError({required this.code, required this.message});
}
